lib/google/gax/api_call.rb

```diff
@@ -45,74 +46,63 @@ class ApiCall
       # The result is another proc which has the same signature as the original.
       #
       # @param stub_method [Proc] Used to make a bare rpc call.
-      # @param timeout [Numeric] client-side Timeout for API calls.
-      # @param metadata [Hash] Request headers.
-      # @param retry_policy [Hash] The settings for error retry, will be merged to the {CallOptions#retry_policy} object
-      #   if supported.
-      # @param params_extractor [Proc] Extracts routing header params from the request.
-      # @param exception_transformer [Proc] If an API exception occurs this transformer is given the original exception
-      #   for custom processing instead of raising the error directly.
       #
-      def initialize stub_method, timeout: nil, metadata: nil, retry_policy: nil, params_extractor: nil,
-                     exception_transformer: nil
-        @stub_method           = stub_method
-        @timeout               = timeout
-        @metadata              = metadata
-        @retry_policy          = retry_policy
-        @params_extractor      = params_extractor
-        @exception_transformer = exception_transformer
+      def initialize stub_method
+        @stub_method = stub_method
       end
 
       ##
       # Invoke the API call.
       #
       # @param request [Object] The request object.
-      # @param options [CallOption, Hash] The options for making the API call.
-      # @param block [Proc] The proc to call when the API call is made.
+      # @param options [ApiCall::Options, Hash] The options for making the API call. A Hash can be provided to customize
+      #   the options object, using keys that match the arguments for {ApiCall::Options.new}. This object should only be
+      #   used once.
+      # @param on_response [Proc] A Proc object to format the response object. The Proc should accept response as an
```

Comment:
Could do this during the doc overhaul, but a little more guidance on how each of these `on_*` blocks are used, if they are mutually exclusive or not, etc. would help.

Reply:
Agreed. Pairing with @quartzmo on the docs will be very beneficial, as he is great at pulling info out and communicating it clearly.

Usage is demonstrated in the tests.

Reply:
I have added some code examples for now. PTAL.